tests/Feature/SampleFeatureTest.php

```diff
@@ -31,4 +38,288 @@ public function testWithMock()
         self::assertEquals(1, $user1['id']);
         self::assertEquals(2, $user2['id']);
     }
+
+    /**
+     * Test sorting by single column.
+     *
+     * @param Collection $resultData Collection of data
+     * @param string $sortingField Sorting field
+     * @param \Exception|null $expectedException Expected exception
+     *
+     * @dataProvider sortingData
+     */
+    public function testSingleSorting(Collection $resultData, string $sortingField, $expectedException = null)
+    {
+        if ($expectedException) {
+            $this->expectException($expectedException);
+        }
+
+        $serviceMock = Mockery::mock(MyService::class);
+        $serviceMock->shouldReceive('getData')->withArgs([$sortingField])->andReturn($resultData);
+        $serviceMock->shouldReceive('getData')->withArgs(["-".$sortingField])
+            ->andReturn($resultData->reverse());
+
+        $this->app->bind(MyService::class, function () use ($serviceMock) {
+            return $serviceMock;
+        });
+
+        $count = $resultData->count();
+
+        $this->assertSortingWorks("api/test-order-by", $count, [$sortingField], []);
+    }
+
+    /** Data for testing sorting by single field */
+    public function sortingData()
```

Comment:
Better give self-descriptive names to all methods, including dataProviders for data-driven tests. Ex. `sortBySingleFieldDataExamples()`

Reply:
done.